java/src/main/java/io/rapidpro/flows/runner/Step.java

```diff
@@ -107,11 +116,21 @@ public void addActionResult(Action.Result actionResult) {
         }
     }
 
+    public Flow getFlow() {
+        return m_flow;
+    }
+
     public List<String> getErrors() {
         return m_errors;
     }
 
     public boolean isCompleted() {
-        return m_leftOn != null;
+        return m_leftOn != null || m_terminal;
```

Comment:
So `m_terminal` is needed because `m_leftOn` is null when we leave subflow? Would it work to set `m_leftOn` when returning to parent? That seems more "correct" since we have actually left that step.